### ISSUE TITLE
`modernize-use-equals-delete` should not warn if all other function declarations are private

### ISSUE THREAD

**Issue by philnik777** [llvm/llvm-project#54276]
`modernize-use-equals-delete` warns that deleted member functions should be public, even when all other member function declarations aren't.
```
class S {
    void do_stuff(int);
    void do_stuff(long) = delete; // modernize-use-equals-delete warns here

protected:
    void do_other_stuff(int);
    void do_other_stuff(long) = delete; // and here
};
```
https://godbolt.org/z/YGojEGa1s

**Comment by FStefanni** [comment]
Hi,

a small update:
I do not know if this is related to this issue, or if it is a VS Code plugin issue, **but** ...
I tried to silence the warning, without success:
```c++
class App
{
public:
App & getInstance();

private:
    App();
    ~App() override;

    // NOLINTBEGIN(*-use-equals-delete)
    App(const App &) = delete;                      // warn
    App(App &&) = delete;                             // warn
    App & operator =(const App &) = delete; // warn
    App & operator =(App &&) = delete;        // warn
    // NOLINTEND(*-use-equals-delete)
};
```

Regards.

**Comment by njames93** [comment]
@FStefanni that's a separate issue that should be fixed in https://reviews.llvm.org/D119701

**Comment by chrchr-github** [comment]
This still happens in clang19/trunk (see link in OP).

**Comment by duncanthomson** [comment]
Related: https://github.com/llvm/llvm-project/issues/135249
